src/ui/marker.js

```diff
@@ -479,19 +487,22 @@ export default class Marker extends Evented {
     _calculateXYTransform(): string {
         const pos = this._pos;
         const map = this._map;
-
-        if (this.getPitchAlignment() !== 'map' || !map || !pos) { return ''; }
-        if (!map._usingGlobe()) {
-            const pitch = map.getPitch();
-            return pitch ? `rotateX(${pitch}deg)` : '';
+        const alignment = this.getPitchAlignment();
+
+        if (map && pos && alignment === 'map') {
+            if (!map._usingGlobe()) { // 'map' alignment on a flat map
+                const pitch = map.getPitch();
+                return pitch ? `rotateX(${pitch}deg)` : '';
+            } // "map" alignment on globe
+            const tilt = radToDeg(globeTiltAtLngLat(map.transform, this._lngLat));
+            const posFromCenter = pos.sub(globeCenterToScreenPoint(map.transform));
+            const tiltOverDist =  tilt / (Math.abs(posFromCenter.x) + Math.abs(posFromCenter.y));
+            const yTilt = posFromCenter.x * tiltOverDist;
+            const xTilt = -posFromCenter.y * tiltOverDist;
+            if (!xTilt && !yTilt) { return ''; }
+            return `rotateX(${xTilt}deg) rotateY(${yTilt}deg)`;
         }
-        const tilt = radToDeg(globeTiltAtLngLat(map.transform, this._lngLat));
-        const posFromCenter = pos.sub(globeCenterToScreenPoint(map.transform));
-        const tiltOverDist =  tilt / (Math.abs(posFromCenter.x) + Math.abs(posFromCenter.y));
-        const yTilt = posFromCenter.x * tiltOverDist;
-        const xTilt = -posFromCenter.y * tiltOverDist;
-        if (!xTilt && !yTilt) { return ''; }
-        return `rotateX(${xTilt}deg) rotateY(${yTilt}deg)`;
+        return ''; // Invalid alignments behave as viewport
```

Comment:
A few comments to improve clarity for that function:
1. Do error checks first instead of last, the validation of inputs is the most important thing when reading what a function does. e.g. 
```
function() {
    if (bad_arguments || error) {
        return '';
    }
    ...
}
```
2. For checks that aren't trivial (e.g. one liner), try to not return in the middle of the function to consolidate the control flow to only a few return statements, so that it's easier to follow what the function does. Otherwise it makes it really hard to read and can be a source of bugs when new statements are added.
```
        if (map && pos && alignment === 'map') {
            if (!map._usingGlobe()) { // 'map' alignment on a flat map
                const pitch = map.getPitch();
                return pitch ? `rotateX(${pitch}deg)` : '';
            } // "map" alignment on globe
            const tilt = radToDeg(globeTiltAtLngLat(map.transform, this._lngLat));
            const posFromCenter = pos.sub(globeCenterToScreenPoint(map.transform));
            const tiltOverDist =  tilt / (Math.abs(posFromCenter.x) + Math.abs(posFromCenter.y));
            const yTilt = posFromCenter.x * tiltOverDist;
            const xTilt = -posFromCenter.y * tiltOverDist;
            if (!xTilt && !yTilt) { return ''; }
            return `rotateX(${xTilt}deg) rotateY(${yTilt}deg)`;
        }
        return ''; // Invalid alignments behave as viewport
```
This block is too dense and has 4 different return statements, we can make the condition more explicit, do error check first and store the result in a variable if needed, something like:
```
function() {
    let result = default;
    if (error) {
        return result;
    }
    
    if (...) {
        result = ...
    } else if (...) {
        result = ...
    }
    return result;
}
```

3. Prefer multi line comment instead of same line comments, e.g.:
```
if (!map._usingGlobe()) {
    // 'map' alignment on a flat map
    ...
}
// "map" alignment on globe
```
instead of
```
if (!map._usingGlobe()) { // 'map' alignment on a flat map
    ...
}  // "map" alignment on globe
```